### PR TITLE
[#1517] Set default file size during resource upload instead of empty size

### LINKED ISSUE #1517: Empty mime type and size for uploaded files
Are these supposed to be filled?

### PR DESCRIPTION
**Pull request by gleb-rudenko**
Fixes #1517

### Proposed fixes:
Default file size is added to the data_dict (if no user file size provided from API ) from the file_upload object that is provided by ResourceUpload Class, during resource_create action.

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckan/logic/action/create.py [hunk 1]
```diff
@@ -296,6 +296,12 @@ def resource_create(context, data_dict):
 
     upload = uploader.get_resource_uploader(data_dict)
 
+    if not 'size' in data_dict and hasattr(upload, 'upload_file'):
```
Comment (TkTech) [inline ckan/logic/action/create.py]: `if 'size' not in data_dict...`

Comment (gleb-rudenko) [inline ckan/logic/action/create.py]: @TkTech , I tried to follow the same code style as in the CKAN code.
From the same method above my code:
`if not 'resources' in pkg_dict:`

Comment (gleb-rudenko) [inline ckan/logic/action/create.py]: @TkTech , if you still want me to change it, let me know please and I will.

Comment (TkTech) [inline ckan/logic/action/create.py]: Yes, please. Your IDE/linter should be yelling at you to fix it :) At lot of the old CKAN code does not follow PEP8 and it's cleaned up as we go.

**Comment by gleb-rudenko** [comment]
@TkTech , done, next time I'll know 👍

**Comment by amercader** [comment]
Thanks @gleb-rudenko, there's another PR which is almost done and which among other things has the same functionality: https://github.com/ckan/ckan/pull/3205. The approach seems very similar and it has tests so it probably makes sense to merge that one. Maybe you could check it to review the approach.

Does that sound good?

**Comment by gleb-rudenko** [comment]
@amercader , I've reviewed the approach, it looks good. I agree with merging it instead of mine.
Left a comment there about PEP8 standards.

**Pull request closed without merging**